src/master-playlist-controller.js

```diff
@@ -446,6 +446,22 @@ export default class MasterPlaylistController extends videojs.EventTarget {
     this.audioPlaylistLoader_.start();
   }
 
+
+  /**
+   * Re-tune playback quality level for the current player
+   * conditions. This method may perform destructive actions, like
+   * removing already buffered content, to readjust the currently
+   * active playlist quickly.
+   */
+  fastQualityChange_() {
+    let media = this.hlsHandler.selectPlaylist();
+
+    if (media !== this.masterPlaylistLoader_.media()) {
+      this.masterPlaylistLoader_.media(media);
+      this.mainSegmentLoader_.sourceUpdater_.remove(this.currentTimeFunc() + 5, Infinity);
```

Comment:
And do we want to add a method to mainSegmentLoader for removing certain time ranges?